src/python/pants/rules/core/targets.py

```diff
@@ -1,20 +1,208 @@
+
+    For example, you can use `prep_command()` to execute a script that sets up tunnels to database
+    servers. These tunnels could then be leveraged by integration tests.
+
+    Pants will only execute the `prep_command()` under the specified goal, when processing targets
+    that depend on the `prep_command()` target.  If not otherwise specified, prep_commands
+    execute in the test goal.
+
+    See also the target type jvm_prep_command() for running tasks defined by a JVM language.
+    """
+
+    alias = "prep_command"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        PrepCommandExecutable,
+        PrepCommandArgs,
+        PrepCommandEnviron,
+        PrepCommandGoals,
+    )
+
+
+class RemoteSourcesTargetRequestedAddress(StringField):
+    """The address of the target which specifies the JAR whose sources will be unpacked.
+
+    Usually, this is an `unpacked_jars()` target.
+    """
+
+    alias = "sources_target"
+    required = True
+
+
+class RemoteSourcesTargetType(PrimitiveField):
+    """The target type of the synthetic target to generate.
+
+    Use the raw symbol rather than a string, e.g. `java_library` rather than `"java_library"`.
+    """
+
+    alias = "dest"
+    required = True
+    value: Type[TargetV1]
+
+    @classmethod
+    def compute_value(
+        cls, raw_value: Optional[Type[TargetV1]], *, address: Address
+    ) -> Type[TargetV1]:
+        # TODO: Once we switch over symbols to refer to V2 targets, we will need to convert back
+        # the V2 target to the V1 target before passing this down to the V1 code.
+        return cast(Type[TargetV1], super().compute_value(raw_value, address=address))
+
+
+# TODO: maybe add a DictField? This could be a can of worms. How should we type the values? It
+#  doesn't seem safe to assume `Dict[str, str]`. We would want to use `FrozenDict` to store the
+#  computed value, but how should we handle converting the values into hashable types, e.g.
+#  converting a List[str] to Tuple[str, ...]?
+#
+#  Audit other uses of dict fields before adding something.
+class RemoteSourcesArgs(PrimitiveField):
+    """Any additional arguments necessary to construct the synthetic destination target (sources and
+    dependencies are supplied automatically)."""
+
+    alias = "args"
+    value: Optional[Dict[str, Any]]
+    default = None
+
+    @classmethod
+    def compute_value(
+        cls, raw_value: Optional[Dict[str, Any]], *, address: Address
+    ) -> Optional[Dict[str, Any]]:
+        value_or_default = super().compute_value(raw_value, address=address)
+        if value_or_default is not None and not isinstance(value_or_default, dict):
+            raise InvalidFieldTypeException(
+                address, cls.alias, value_or_default, expected_type="a dictionary"
+            )
+        return value_or_default
+
+
+class RemoteSources(Target):
+    """A V1-only target that generates a synthetic target using deferred sources.
```

Comment:
I can see PrepCommand being ruled out of v2 since part of its semantic - tasks - are gone. This though seems less obvious. Fetching source files from a remote source seems to make perfect sense still in a v2 world. Unless I've missed reasoning on why this won't survive, perhaps drop V1-only verbage.

Reply:
Okay, sounds good! Thanks for the explanation.